js/Multicooker.js

```diff
@@ -0,0 +1,59 @@
+"use strict";
+
+class Multicooker {
+  constructor() {
+    this.__name = null;
+    this.__state = null;
+    this.__tasklist = ["task1", "task2", "task3", "task4"];
+    this.__task = null; //a task for device
+    this.__currentTask = 0; //a task that is currently viewed in task list
+    this.__temperature = 100;
+  }
+
+  tasklist() {
+    for (const item of this.__tasklist) {
+      console.log(item);
+    }
+  }
+
+  get task() {
+    if (this.__task != null) {
```

Comment:
get should always return a value. In your case it is either task or undefined which is buggy